File: backend/microservice/src/level/api/internal/admin/ReviewLevelSubmissionInternalApi.scala
package microservice.level.api.internal.admin

import cats.effect.IO
import java.sql.Connection
import microservice.infrastructure.api.{APIMessage, PlanSteps}
import microservice.infrastructure.http.HttpError
import microservice.level.objects.submission.Submission
import microservice.level.support.admin.SubmissionReviewSupport
import microservice.system.objects.SubmissionStatus

/** 模块间 API：审核关卡投稿；由 admin HTTP API 调用，不挂路由。 */
final case class ReviewLevelSubmissionInternalAPIMessage(
  submissionId: String,
  reviewerId: String,
  status: SubmissionStatus,
  reviewNote: Option[String],
  reviewedAt: String
) extends APIMessage[Submission] {
  override def plan(connection: Connection): IO[Either[HttpError, Submission]] =
    PlanSteps.finish {
      SubmissionReviewSupport.requireReview(connection, submissionId, reviewerId, status, reviewNote, reviewedAt)
    }
}

File: frontend/src/api/level/internal/admin/submissions/ReviewLevelSubmissionInternalApi.ts
import { z } from "zod";
import { APIMessage } from "../../../../../system/api/APIMessage.js";

/** Mirrors backend internal API level/api/internal/admin/submissions/ReviewLevelSubmissionInternalApi.scala; not registered as a public frontend route. */
export class ReviewLevelSubmissionInternalAPI extends APIMessage<unknown> {
  readonly submissionId: unknown;
  readonly reviewerId: unknown;
  readonly status: unknown;
  readonly reviewNote: unknown;
  readonly reviewedAt: unknown;
  constructor(submissionId: unknown, reviewerId: unknown, status: unknown, reviewNote: unknown, reviewedAt: unknown) {
    super();
    this.submissionId = submissionId;
    this.reviewerId = reviewerId;
    this.status = status;
    this.reviewNote = reviewNote;
    this.reviewedAt = reviewedAt;
  }

  override get responseSchema() {
    return z.unknown();
  }
}
